### PR TITLE
LPS-183141 Performace problems in upgrade due to dynamic query use

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-183141

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
ci:test:stable

**Comment by WYZ-Issac** [comment]
The stable failures also appear in other commits, they are not related to your PR.

**Review by marcogalluzzi — COMMENTED** [review]
It looks good to me! Just a comment: isn't better (in terms of good practice) to get the values of a result set by the name instead of the position? I mean using: `resultSet.getLong("bitwiseValue")` instead of `resultSet.getLong(1)`

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
It looks good to me! Just a comment: isn't better (in terms of good practice) to get the values of a result set by the name instead of the position? I mean using: `resultSet.getLong("bitwiseValue")` instead of `resultSet.getLong(1)`

File: portal-kernel/src/com/liferay/portal/kernel/upgrade/BaseViewActionResourcePermissionUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				"select bitwiseValue from ResourceAction where name = ? and " +
+					"actionId = ?")) {
+
+			preparedStatement.setString(1, getClassName());
+			preparedStatement.setString(2, getActionId());
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			if (resultSet.next()) {
+				return resultSet.getLong(1);
```
Comment (marcogalluzzi) [inline portal-kernel/src/com/liferay/portal/kernel/upgrade/BaseViewActionResourcePermissionUpgradeProcess.java:78]: You missed that one!

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Review by marcogalluzzi — APPROVED** [review]

**Comment by Hanlf** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:test:stable

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

**Comment by Hanlf** [comment]
ci:forward:force

**Pull request closed without merging**